tests/drum/test_data_marshalling.py

```diff
@@ -27,6 +32,70 @@ def test__order_by_float():
     ]
 
 
-def test_can_be_converted():
-    assert _can_be_converted_to_float(["05.99999999", "0.2", "-.13"])
-    assert not _can_be_converted_to_float(["1.0_", "1", "2"])
+def test_marshal_predictions_multiclass_happy():
+    preds = np.array([[1, 0, 0], [1, 0, 0]])
+    labels = [1, 2, 3]
+    expected = pd.DataFrame(data=preds, columns=labels)
+    pd.testing.assert_frame_equal(
+        marshal_predictions(
+            request_labels=labels, predictions=preds, target_type=TargetType.MULTICLASS
+        ),
+        expected,
+    )
+
+
+def test_marshal_predictions_multiclass_wrong_label_length():
+    preds = np.array([[1, 0, 0], [1, 0, 0]])
+    labels = [1, 2, 3, 4]
+    with pytest.raises(
+        DrumCommonException,
+        match=" predictions must return the probability distribution for all class labels",
+    ):
+        marshal_predictions(
+            request_labels=labels, predictions=preds, target_type=TargetType.MULTICLASS
+        )
+
+
+def test_marshal_predictions_binary_reshape_happy():
+    preds = np.array([1, 1, 1])
+    labels = [1, 2]
+    marshal_predictions(request_labels=labels, predictions=preds, target_type=TargetType.BINARY)
+
+
+def test_marshal_predictions_reshape_regression_happy():
+    preds = np.array([1, 1, 1])
+    labels = [1]
+    marshal_predictions(request_labels=labels, predictions=preds, target_type=TargetType.REGRESSION)
+
+
+def test_marshal_predictions_invalid_dtype():
+    preds = np.zeros((1, 2, 3))
+    labels = [1]
+    with pytest.raises(
+        DrumCommonException, match="predictions must return a np array, but received"
+    ):
+        marshal_predictions(
+            request_labels=labels, predictions="elephant", target_type=TargetType.REGRESSION
+        )
+    with pytest.raises(DrumCommonException, match="predictions must return a 2 dimensional array"):
+        marshal_predictions(
+            request_labels=labels, predictions=preds, target_type=TargetType.REGRESSION
+        )
+
+
+def test_marshal_predictions_bad_shape_regression():
+    labels = [1]
+    preds = np.zeros((1, 2))
+    with pytest.raises(DrumCommonException, match="must contain only 1 column"):
+        marshal_predictions(
+            request_labels=labels, predictions=preds, target_type=TargetType.REGRESSION
+        )
+
+
+def test_marshal_predictions_dont_add_to_one():
```

Comment:
please add a test for invalid other ways of adding to 1 (ex. 2, -1)

Reply:
how come this is bad?

Reply:
because 2, -1 isn't a valid probability distribution but it still sums to 1

Reply:
what is invalid about it? the fact there is a negative? I can assert no negatives if you'd like

Reply:
Yes, as insinuated in my previous comment, -1 is not a valid probability. Probabilities should be in [0.0, 1.0]